packages/mobile/src/fiatExchanges/SimplexScreen.tsx

```diff
@@ -0,0 +1,170 @@
+import { showError } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { numberVerifiedSelector } from 'src/app/selectors'
+import BackButton from 'src/components/BackButton'
+import WebView from 'src/components/WebView'
+import { CurrencyCode } from 'src/config'
+import ReviewFees from 'src/fiatExchanges/ReviewFees'
+import { fetchSimplexPaymentRequest, generateSimplexCheckoutForm } from 'src/fiatExchanges/utils'
+import { CURRENCY_ENUM } from 'src/geth/consts'
+import i18n from 'src/i18n'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import { emptyHeader, HeaderTitleWithBalance } from 'src/navigator/Headers'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { navigateToURI } from 'src/utils/linking'
+import { currentAccountSelector } from 'src/web3/selectors'
+
+export const simplexOptions = () => ({
+  ...emptyHeader,
+  headerLeft: () => <BackButton />,
+  headerTitle: () => <HeaderTitleWithBalance title={i18n.t('fiatExchangeFlow:addFunds')} />,
+})
+
+type RouteProps = StackScreenProps<StackParamList, Screens.Simplex>
+type Props = RouteProps
+
+function SimplexScreen({ route, navigation }: Props) {
+  const [loadSimplexCheckout, setLoadSimplexCheckout] = useState(false)
+  const [redirected, setRedirected] = useState(false)
+
+  const { simplexQuote, userIpAddress } = route.params
+
+  const account = useSelector(currentAccountSelector)
+  const e164PhoneNumber = useSelector(e164NumberSelector)
+  const phoneNumberConfirmed = useSelector(numberVerifiedSelector)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+
+  const onNavigationStateChange = ({ url }: any) => {
+    if (url?.startsWith('http')) {
+      setRedirected(true)
+    }
+    if (url?.startsWith('celo://wallet')) {
+      navigateToURI(url)
+    }
+  }
+
+  const onButtonPress = () => {
+    setLoadSimplexCheckout(true)
+    navigation.setOptions({
+      ...emptyHeader,
+      headerLeft: () => <BackButton />,
+      headerTitle: 'Simplex',
+    })
+  }
+
+  useLayoutEffect(() => {
+    const token =
+      simplexQuote.digital_money.currency.toLowerCase() === 'cusd'
+        ? CURRENCY_ENUM.DOLLAR
+        : CURRENCY_ENUM.GOLD
+    navigation.setOptions({
+      ...emptyHeader,
+      headerLeft: () => <BackButton />,
+      headerTitle: () => (
+        <HeaderTitleWithBalance title={i18n.t('fiatExchangeFlow:addFunds')} token={token} />
+      ),
+    })
+  }, [])
+
+  const asyncSimplexPaymentRequest = useAsync(async () => {
+    if (!account) {
+      return
+    }
+    return fetchSimplexPaymentRequest(
+      account,
+      e164PhoneNumber,
+      phoneNumberConfirmed,
+      simplexQuote,
+      userIpAddress
+    )
+  }, [])
+
+  const simplexPaymentRequest = asyncSimplexPaymentRequest?.result
+
+  if (asyncSimplexPaymentRequest.status === 'error') {
+    showError(ErrorMessages.SIMPLEX_PURCHASE_FETCH_FAILED)
```

Comment:
Won't this call `showError` on every render? Should this be inside a `useEffect` or something?